lib/dataLayer.tsx

```diff
@@ -510,10 +512,151 @@ async function _submitToAPI(values: Responses, formikBag: FormikBag<DynamicFormP
     });
 }
 
+// CRUD for Organisations
+async function _crudOrganisationsWithCache(
+  payload: CrudOrganisationInput
+): Promise<CrudOrganisationResponse> {
+  if (payload.method === "GET" && payload.organisationID) {
+    const cachedValue = await organisationCache.organisationID.check(payload.organisationID);
+    if (cachedValue) {
+      return cachedValue;
+    }
+  }
+
+  return await _crudOrganisations(payload).then((response) => {
+    switch (payload.method) {
+      case "GET":
+        if (payload.organisationID) {
+          organisationCache.organisationID.set(payload.organisationID, response);
+        }
+        break;
+      case "DELETE":
+      case "UPDATE":
+      case "INSERT":
+        if (payload.organisationID) {
+          organisationCache.organisationID.invalidate(payload.organisationID);
+        }
+        break;
+      default:
+        break;
+    }
+
+    return response;
+  });
+}
+
+async function _crudOrganisations(
+  payload: CrudOrganisationInput
+): Promise<CrudOrganisationResponse> {
+  const getConfig = (payload: CrudOrganisationInput) => {
+    const { method, organisationID, organisationNameEn, organisationNameFr } = payload;
+
+    switch (payload.method) {
+      case "GET":
+        return {
+          method,
+          organisationID,
+        };
+      case "INSERT":
+        return {
+          method,
+          organisationNameEn,
+          organisationNameFr,
+        };
+      case "UPDATE":
+        return {
+          method,
+          organisationID,
+          organisationNameEn,
+          organisationNameFr,
+        };
+      case "DELETE":
+        return {
+          method: "DELETE",
+          organisationID,
+        };
+      default:
+        return {
+          method: "UNDEFINED",
+        };
+    }
+  };
+
+  if (process.env.CYPRESS && payload.method !== "GET") {
+    logMessage.info("Organisations CRUD API in Test Mode");
+    const timer = () => {
+      return new Promise((resolve) => {
+        setTimeout(() => {
+          resolve(undefined);
+        }, 1000);
+      });
+    };
+
+    return timer().then(() => {
+      const { method } = payload;
+      switch (method) {
+        case "INSERT":
+          return {
+            data: {
+              records: [
+                {
+                  organisationID: "11111111-1111-1111-1111-111111111111",
+                  organisationNameEn: "Test",
+                  organisationNAmeFr: "Test",
+                },
+              ],
+            },
+          };
+        case "UPDATE":
+          return {
+            data: {},
+          };
+        case "DELETE":
+          return {
+            data: {},
+          };
+        default:
+          return {
+            data: {},
+          };
+      }
+    });
+  }
+
+  const lambdaClient = new LambdaClient({
```

Comment:
Just a placeholder but performance testing showed that we should pull the LambdaClient initialisations out of the function level and up to the module level.  Out of scope for this PR but just an FYI.  We'll be able to reuse the same `lambdaClient` for all the invocations in this file.